### PR TITLE
Add more impl_nvfuser for prims

### PR DESCRIPTION
**Pull request by IvanYashchuk**
This PR adds `test_nvfuser_impl_is_used` that checks that the corresponding nvfuser op (if available) is used in the prim definition.

Adds `impl_nvfuser=` for atan2, bitwise_and, bitwise_or, bitwise_xor, eq, ne, pow, sub, sum, where, rsqrt, lgamma.

### CODE REVIEW

File: torch/_prims/__init__.py [hunk 1]
```diff
@@ -901,45 +964,28 @@ def _div_nvfuser(fd: Any, a: TensorLikeType, b: TensorLikeType):
...
 
-
-def _fmax_nvfuser(fd: Any, a: TensorLikeType, b: TensorLikeType):
-    return fd.Ops.fmax(a, b)  # type: ignore[attr-defined]
-
-
 fmax = _make_elementwise_binary_prim(
     "fmax",
     impl_aten=torch.fmax,
-    impl_nvfuser=_fmax_nvfuser,
```
Comment (IvanYashchuk) [inline torch/_prims/__init__.py:920]: Unfortunately, there are no `fmax`, `fmin` ops in nvfuser yet.

File: torch/_prims/__init__.py [hunk 2]
```diff
@@ -1094,13 +1116,7 @@ def _mul_nvfuser(fd: Any, a: TensorLikeType, b: TensorLikeType):
 pow = _make_elementwise_binary_prim(
     "pow",
     impl_aten=torch.pow,
-    doc="",
-    type_promotion=ELEMENTWISE_PRIM_TYPE_PROMOTION_KIND.DEFAULT,
-)
-
-rsqrt = _make_elementwise_binary_prim(
```
Comment (IvanYashchuk) [inline torch/_prims/__init__.py:1101]: `rsqrt` is not a binary prim, but unary. How can we prevent these mistakes in the future?

File: torch/_prims/__init__.py [hunk 3]
```diff
@@ -173,6 +173,51 @@
...
+    "le",
+    "lt",
+    "mul",
+    "ne",
+    "pow",
+    "sub",
+}
+
+for fname in _nvfuser_binary_ops:
+    exec(
```
Comment (IvanYashchuk) [inline torch/_prims/__init__.py:196]: Is `exec` with f-strings fine to use?

Comment (IvanYashchuk) [reply torch/_prims/__init__.py:196]: Nope.
linrunner doesn't like that 😕 
```
2022-05-30T16:00:02.3194762Z ##[error]Name "_eq_nvfuser" is not defined 
2022-05-30T16:00:02.3196027Z ##[error]Name "_fmod_nvfuser" is not defined 
2022-05-30T16:00:02.3197286Z ##[error]Name "_ge_nvfuser" is not defined 
2022-05-30T16:00:02.3198566Z ##[error]Name "_gt_nvfuser" is not defined 
2022-05-30T16:00:02.3199834Z ##[error]Name "_le_nvfuser" is not defined 
2022-05-30T16:00:02.3201117Z ##[error]Name "_lt_nvfuser" is not defined 
2022-05-30T16:00:02.3202396Z ##[error]Name "_mul_nvfuser" is not defined 
2022-05-30T16:00:02.3203702Z ##[error]Name "_ne_nvfuser" is not defined 
2022-05-30T16:00:02.3204976Z ##[error]Name "_pow_nvfuser" is not defined 
2022-05-30T16:00:02.3206247Z ##[error]Name "_sub_nvfuser" is not defined 
2022-05-30T16:00:02.3207533Z ##[error]Name "_where_nvfuser" is not defined 
```

Comment (IvanYashchuk) [reply torch/_prims/__init__.py:196]: Can be silenced with `ignore[name-defined]`.

File: torch/_prims/__init__.py (hunk 2 above)
Comment (mruberry) [reply torch/_prims/__init__.py:1101]: We just need to add a reference for it so it's tested (currently we're only testing refs directly and prims indirectly through them)

Comment (mruberry) [reply torch/_prims/__init__.py:1101]: We just need to add a reference for it so it's tested (currently we're only testing refs directly and prims indirectly through them)

**Review by mruberry — APPROVED** [review]
Stamped!

**Pull request closed without merging**